tspi/nv.go

```diff
@@ -26,8 +26,13 @@ type NV struct {
 
 // ReadValue reads length bytes from offset in the TPM NVRAM space
 func (nv *NV) ReadValue(offset uint, length uint) ([]byte, error) {
-	data := make([]byte, length)
-	err := tspiError(C.Tspi_NV_ReadValue(nv.handle, (C.UINT32)(offset), (*C.UINT32)(unsafe.Pointer(&length)), (**C.BYTE)(unsafe.Pointer(&data))))
+	var cdata *C.BYTE
+	defer C.Tspi_Context_FreeMemory(nv.context, cdata)
+	err := tspiError(C.Tspi_NV_ReadValue(nv.handle, (C.UINT32)(offset), (*C.UINT32)(unsafe.Pointer(&length)), &cdata))
+	if err != nil {
+		return nil, err
+	}
+	data := C.GoBytes(unsafe.Pointer(cdata), C.int(length))
```

Comment:
Does this actually create a copy of the cdata?

Reply:
Yep, according to [](https://golang.org/cmd/cgo/) it should

> A few special functions convert between Go and C types by making copies of the data. In pseudo-Go definitions:
```
// C data with explicit length to Go []byte
func C.GoBytes(unsafe.Pointer, C.int) []byte
```